apps/emqx/src/emqx_cm.erl

```diff
@@ -455,60 +465,62 @@ discard_session(ClientId) when is_binary(ClientId) ->
 when
     Action :: kick | discard | {takeover, 'begin'} | {takeover, 'end'} | takeover_kick.
 request_stepdown(Action, ConnMod, Pid) ->
-    Timeout =
-        case Action == kick orelse Action == discard of
-            true -> ?T_KICK;
-            _ -> ?T_TAKEOVER
-        end,
+    Timeout = ?IF_KILL_ACTION(Action, ?T_KICK, ?T_TAKEOVER),
     Return =
-        %% this is essentially a gen_server:call implemented in emqx_connection
-        %% and emqx_ws_connection.
-        %% the handle_call is implemented in emqx_channel
         try apply(ConnMod, call, [Pid, Action, Timeout]) of
             ok -> ok;
             Reply -> {ok, Reply}
         catch
-            % emqx_ws_connection: call
-            _:noproc ->
-                ok = ?tp(debug, "session_already_gone", #{stale_pid => Pid, action => Action}),
-                {error, noproc};
-            % emqx_connection: gen_server:call
-            _:{noproc, _} ->
-                ok = ?tp(debug, "session_already_gone", #{stale_pid => Pid, action => Action}),
-                {error, noproc};
-            _:{shutdown, _} ->
-                ok = ?tp(debug, "session_already_shutdown", #{stale_pid => Pid, action => Action}),
-                {error, noproc};
-            _:{{shutdown, _}, _} ->
-                ok = ?tp(debug, "session_already_shutdown", #{stale_pid => Pid, action => Action}),
-                {error, noproc};
-            _:{timeout, {gen_server, call, _}} ->
-                ?tp(
-                    warning,
-                    "session_stepdown_request_timeout",
-                    #{stale_pid => Pid, action => Action, stale_channel => stale_channel_info(Pid)}
-                ),
-                ok = force_kill(Pid),
-                {error, timeout};
-            _:Error:St ->
-                ?tp(
-                    error,
-                    "session_stepdown_request_exception",
-                    #{
-                        stale_pid => Pid,
-                        action => Action,
-                        reason => Error,
-                        stacktrace => St,
-                        stale_channel => stale_channel_info(Pid)
-                    }
-                ),
-                ok = force_kill(Pid),
-                {error, unexpected_exception}
+            Err:Reason:St ->
+                handle_stepdown_exception(Err, Reason, St, ConnMod, Pid, Action)
         end,
-    case Action == kick orelse Action == discard of
-        true -> ok;
-        _ -> Return
-    end.
+    ?IF_KILL_ACTION(Action, ok, Return).
```

Comment:
I don't think this is actually making the code more readable.

Reply:
this code this duplicated